tokio-quiche/src/quic/router/multipath.rs

```diff
@@ -0,0 +1,701 @@
+                    .path_rxs
+                    .iter()
+                    .map(|(_, addr)| *addr)
+                    .collect(),
+                peer_addr,
+                packet_scheduler: self.config.packet_scheduler.clone(),
+            },
+        );
+
+        conn.audit_log_stats.set_transport_handshake_start(
+            crate::quic::router::instant_to_system(handshake_start_time),
+        );
+
+        self.conns.insert(scid, &conn);
+
+        // Add the client-generated "pending" connection ID to the map as well.
+        if let Some(pending_cid) = pending_cid {
+            self.conns.map_cid(pending_cid, &conn);
+        }
+
+        self.metrics.accepted_initial_packet_count().inc();
+        if self.config.enable_expensive_packet_count_metrics {
+            if let Some(peer_ip) =
+                quic_expensive_metrics_ip_reduce(conn.peer_addr().ip())
+            {
+                self.metrics
+                    .expensive_accepted_initial_packet_count(peer_ip)
+                    .inc();
+            }
+        }
+
+        send_permit.send(Ok(conn));
+        Ok(())
+    }
+}
+
+impl<Tx, Rx, M, I> MultiPathInboundRouter<Tx, Rx, M, I>
+where
+    Tx: DatagramSocketSend + Send + Sync + 'static,
+    Rx: DatagramSocketRecv,
+    M: Metrics,
+    I: InitialPacketHandler,
+{
+    // Poll a single path for incoming data
+    fn poll_path(
+        &mut self, cx: &mut Context<'_>, path_id: usize,
+    ) -> Poll<io::Result<PollRecvMultiData>> {
+        #[cfg(not(target_os = "linux"))]
+        {
+            // Simple polling for non-Linux platforms
+            let (ref mut rx, local_addr) = self.path_rxs[path_id];
+            let mut buf = tokio::io::ReadBuf::new(&mut self.buffers[path_id]);
+
+            match rx.poll_recv_from(cx, &mut buf) {
+                Poll::Ready(Ok(peer_addr)) => {
+                    let bytes = buf.filled().len();
+                    if bytes == 0 {
+                        return Poll::Ready(Ok(None)); // No data
+                    }
+
+                    let mut buf = std::mem::replace(
+                        &mut self.buffers[path_id],
+                        BufFactory::get_max_buf(),
+                    );
+                    buf.truncate(bytes);
+
+                    Poll::Ready(Ok(Some(Incoming {
+                        peer_addr,
+                        local_addr,
+                        buf,
+                        rx_time: None,
+                        gro: None,
+                        path_id: Some(path_id as u64), // Include path ID
+                    })))
+                },
+                Poll::Ready(Err(e)) => Poll::Ready(Err(e)),
+                Poll::Pending => Poll::Pending,
+            }
+        }
+
+        #[cfg(target_os = "linux")]
+        {
+            use nix::errno::Errno;
+            use nix::sys::socket::*;
+            use std::net::SocketAddrV4;
+            use std::net::SocketAddrV6;
+            use std::os::fd::AsRawFd;
```

Comment:
As it is only useful in such OS.
```suggestion
            use std::os::fd::AsRawFd;
            use std::task::ready;
```